source/materials.c

```diff
@@ -93,6 +93,11 @@ int aws_cryptosdk_keyring_on_encrypt(
     struct aws_array_list *edks,
     const struct aws_hash_table *enc_ctx,
     enum aws_cryptosdk_alg_id alg) {
+    AWS_PRECONDITION(aws_cryptosdk_keyring_is_valid(keyring) && (keyring->vtable != NULL));
+    AWS_PRECONDITION(aws_byte_buf_is_valid(unencrypted_data_key));
```

Comment:
`AWS_PRECONDITION(request_alloc == NULL || aws_allocator_is_valid(request_alloc))`

Reply:
I added the precondition.